Docs about using 3rd party databases (#243)
SFDO-Tooling/Snowfakery: docs/index.md

$ cci flow run test_everything
```

## Using Snowfakery with Databases

Snowfakery is built on top of a very flexible engine called
SQLAlchemy. This allows it to connect to many different databases
subject to the limitations described below.

You should start by installing Snowfakery in a context which
makes it easy to use the Python command 'pip' to manage your
Python environment. For example you could install Python
using the standard installers from `python.org` and then

> you would do these commands to create and use a venv with the

Reviewer: ```suggestion
you would run the following commands to create and use a venv with the
```

Postgres plugin:

```bash
# create a new directory for our experiment
$ mkdir experiment_with_postgres
# cd into it
$ cd experiment_with_postgres
# create a new database: 
# https://www.postgresql.org/docs/9.1/app-createdb.html
$ createdb snowfakerydb
# create a virtual environment. A good habit to get into.
# https://docs.python.org/3/library/venv.html
$ python3 -m venv myvenv
# activate the venv
$ source myvenv/bin/activate
# install Snowfakery in this venv
$ pip install snowfakery
# install the Postgres library for Python
# https://pypi.org/project/psycopg2/
$ pip install psycopg2
# let's use it!
$ snowfakery --dburl='postgresql://localhost:5432/snowfakerydb' ~/code/Snowfakery/examples/company.yml --target-number 1000 Employee
# and check the results
# https://www.postgresql.org/docs/9.3/app-psql.html
$ echo 'select * from "Employee"' | psql snowfakerydb
```

That's a lot to take in, but hopefully it will be clear enough
to follow the links and understand the details.

A limitation of this process is that currently Snowfakery can
only create new tables rather than import into existing ones.

> The table will have an id column and columns for every field that

Reviewer: Wording is slightly confusing at first. Maybe
```suggestion
The table will have an id column, in addition to columns for every field that
```
